oci/pack.go

```diff
@@ -394,6 +396,10 @@ func NewPackageFromTarget(ctx context.Context, targ target.Target, opts ...packm
 		return nil, fmt.Errorf("could not add manifest to index: %w", err)
 	}
 
+	for k, v := range ocipack.labels {
+		ocipack.manifest.SetLabel(ctx, k, v)
```

Comment:
Additionally, these lines need to happen _before_ the `AddManifest` method is called:
```diff
diff --git a/oci/pack.go b/oci/pack.go
index 2f31aed8..97fbdd09 100644
--- a/oci/pack.go
+++ b/oci/pack.go
@@ -392,14 +392,17 @@ func NewPackageFromTarget(ctx context.Context, targ target.Target, opts ...packm
                }
        }
 
-       if err := ocipack.index.AddManifest(ctx, ocipack.manifest); err != nil {
-               return nil, fmt.Errorf("could not add manifest to index: %w", err)
-       }
-
        for k, v := range ocipack.labels {
+               log.G(ctx).
+                       WithField(k, v).
+                       Trace("label")
                ocipack.manifest.SetLabel(ctx, k, v)
        }
 
+       if err := ocipack.index.AddManifest(ctx, ocipack.manifest); err != nil {
+               return nil, fmt.Errorf("could not add manifest to index: %w", err)
+       }
+
        if _, err = ocipack.index.Save(ctx, ocipack.ref.String(), nil); err != nil {
                return nil, fmt.Errorf("could not save index: %w", err)
        }
```

Reply:
yeah, completely missed it